xaitk_saliency/utils/tools/gen_detector_sal.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+from typing import Optional, Collection, Iterable, Tuple, Dict, Hashable
+
+from smqtk_detection.interfaces.detect_image_objects import DetectImageObjects
+from smqtk_detection.utils.bbox import AxisAlignedBoundingBox
+
+from xaitk_saliency.interfaces.gen_detector_prop_sal import GenerateDetectorProposalSaliency
+from xaitk_saliency.interfaces.perturb_image import PerturbImage
+from xaitk_saliency.utils.masking import occlude_image_batch
+from xaitk_saliency.utils.detection import format_detection
+
+
+def gen_detector_sal(
+    ref_imgs: Collection[np.ndarray],
+    detector: DetectImageObjects,
+    img_perturber: PerturbImage,
+    sal_generator: GenerateDetectorProposalSaliency,
+    verbose: Optional[bool] = False,
+) -> Collection[np.ndarray]:
+    """
+    Generate saliency maps for detections in each of the given images.
+    Detections generation, image perturbation, and saliency map creation is
+    performed by the implemenations of ``DetectImageObjects``,
+    ``PerturbImage``, and ``GenerateDetectorProposalSaliency`` that are passed,
+    respectively.
+
+    The resulting saliency maps matrix for each image has a shape of
+    [nDets, imgHeight, imgWidth].
+
+    :param ref_imgs: List of images to generate saliency maps for.
+    :param detector: Implementation of ``DetectImageObjects`` to generate
+        detections with.
+    :param img_perturber: Implementation of ``PerturbImage`` to use.
+    :param sal_generator: Implementation of
+        ``GenerateDetectorProposalSaliency`` to use.
+    :param verbose: Option to print out progress statements. Default is false.
+
+    :returns: List of saliency map matrices, each with shape
+        [nDets, imgHeight, imgWidth] respective to their corresponding image.
+    """
+
+    def dets_to_formatted_mat(
```

Comment:
Does this method need to be defined in-line? I don't see it using any parent-scope values. If not, it would be more appropriate to define this method on the module scope, maybe giving it a prefix `_` to mark it as "private". This may also help with the complexity code-smell that SonarCloud is attributing to the parent function.